extensions/amp-story/0.1/amp-story.js

```diff
@@ -574,7 +574,7 @@ export class AmpStory extends AMP.BaseElement {
       [this.pages_[0]];
 
     const storyLoadPromise = Promise.all(
-        pagesToWaitFor.map(page => page.whenLoaded()));
+        pagesToWaitFor.filter(page => !!page).map(page => page.whenLoaded()));
```

Comment:
I got curious and just learned this so I thought I'd share:
You can actually pass non promise arguments to ``Promise.all`` and they will get ignored, but the returned promise will be fulfilled with an array containing all the values. So your implementation is the best one since storyLoadPromise will resolve with an array of pages, and no falsy values.

Reply:
😄 

It's also needed because we call `.whenLoaded()` on the page, so if the page provided is not actually a page, it will throw a JS error.

Reply:
Yeah, I was thinking

``
pagesToWaitFor.map(page => page && page.whenLoaded()))
``

vs

``
pagesToWaitFor.filter(page => !!page).map(page => page.whenLoaded()))
``